test/extended/util/image_helpers.go

```diff
@@ -18,8 +23,65 @@ func DumpAndReturnTagging(tags []string) ([]string, error) {
 	return hexIDs, nil
 }
 
+// ParseImageStreamImageName splits a string into its name component and ID component, and returns an error
+// if the string is not in the right form.
+func ParseImageStreamImageName(input string) (name string, id string, err error) {
+	segments := strings.SplitN(input, "@", 3)
+	switch len(segments) {
+	case 2:
+		name = segments[0]
+		id = segments[1]
+		if len(name) == 0 || len(id) == 0 {
+			err = fmt.Errorf("image stream image name %q must have a name and ID", input)
+		}
+	default:
+		err = fmt.Errorf("expected exactly one @ in the isimage name %q", input)
+	}
+	return
+}
+
 //CreateResource creates the resources from the supplied json file (not a template); ginkgo error checking included
 func CreateResource(jsonFilePath string, oc *CLI) error {
 	err := oc.Run("create").Args("-f", jsonFilePath).Execute()
 	return err
 }
+
+func GetTagEvents(tagEventList []imagev1.NamedTagEventList, tagName string) []imagev1.TagEvent {
+	result := []imagev1.TagEvent{}
+	for i := range tagEventList {
+		if tagEventList[i].Tag == tagName {
+			continue
+		}
+		result = append(result, tagEventList[i].Items...)
+	}
+	return result
+}
+
+// ImageWithMetadata mutates the given image. It parses raw DockerImageManifest data stored in the image and
+// fills its DockerImageMetadata and other fields.
+// Copied from github.com/openshift/image-registry/pkg/origin-common/util/util.go
+func ImageWithMetadataOrDie(image *imagev1.Image) {
```

Comment:
we moved this to `oc`, but if we have another need, move to librar-go

Reply:
i was not sure if e2e test justifies move to library-go